src/components/calendar-bookings/CalendarBookingsForm.tsx

```diff
@@ -0,0 +1,209 @@
+        ...currentState,
+        isConfirmingBooking: false,
+        confirmingBookingError: error as OneBlinkAppsError,
+      }))
+    }
+  }, [postSubmissionAction, onBookingConfirmed])
+
+  return (
+    <>
+      <Collapse in={isConfirmingBooking}>
+        <div className="has-margin-top-1">
+          <div className="cypress-loading has-text-centered">
+            <OnLoading className="has-text-centered" />
+            <Fade in={!!formSubmissionResult?.payment}>
+              <span>Redirecting to payment</span>
+            </Fade>
+          </div>
+        </div>
+      </Collapse>
+
+      <div className="ob-scheduling-booking-form">
+        {!formSubmissionResult?.payment && (
+          <NylasScheduling
+            eventOverrides={{
+              timeslotConfirmed: onTimeSlotConfirmed,
+              bookedEventInfo: async (event) => {
+                event.preventDefault()
+
+                if (event.detail.error) {
+                  setBookingError(
+                    event.detail.error.message ?? 'Calendar Booking Error',
+                  )
+                } else {
+                  event.detail
+                  await handleConfirmedBooking()
+                }
+              },
+            }}
+            bookingInfo={{
+              primaryParticipant: {
+                name: name ?? '',
+                email: email ?? '',
+              },
+              additionalFields: {
+                submissionId: {
+                  value: submissionId,
+                  type: 'text',
+                },
+              },
+            }}
+            enableUserFeedback={false}
+            configurationId={configurationId}
+            sessionId={sessionId}
+            nylasBranding={false}
+          />
+        )}
+
+        {formSubmissionResult && !formSubmissionResult.payment && (
+          <Receipt
+            className="ob-scheduling-receipt"
+            containerClassName="ob-scheduling-receipt__container"
+          >
+            <div className="buttons">
+              <button
+                type="button"
+                className={clsx(
+                  'is-primary button ob-button ob-scheduling-receipt__button ob-scheduling-receipt__okay-button cypress-scheduling-receipt-okay-button',
+                  { 'is-loading': isRunningPostSubmissionAction },
+                )}
+                disabled={isRunningPostSubmissionAction}
+                onClick={() => postSubmissionAction(formSubmissionResult)}
+              >
+                Done
+              </button>
+            </div>
+          </Receipt>
+        )}
+
+        <ErrorModal
+          error={postSubmissionError}
+          onClose={clearPostSubmissionError}
+        />
+
+        <ErrorModal
+          error={confirmingBookingError}
+          onClose={clearConfirmingBookingError}
+        />
+      </div>
+    </>
+  )
+}
+
+function CalendarBookingsForm({
+  postSubmissionAction,
+}: {
+  postSubmissionAction: (
```

Comment:
```suggestion
  onDone: (
```

To align with payment components